source/config/__init__.py

```diff
@@ -642,6 +644,11 @@ def deleteProfile(self, name):
 				if trigger._profile == delProfile:
 					del self._suspendedTriggers[trigger]
 
+	def getActiveProfile(self):

```

Comment:
@feerrenrut  as far as I understand how profiles work on NVDA, there is always the base config (a k a default profile) and zero or one other profile in effect at the same time.

The final configuration is then a merge of all default configurations plus the specified configurations on the profile, which will have precedence over these same configurations on the default profile.

Your comment seems to imply that there is a stack of profiles activated, so every configuration is searched, in precedence order, from the upper to the lower profiles, being the default profile the lowest one.

I frankly have never been able to reproduce this kind of behavior, nor has the graphical user interface ever given me this impression, because, from the GUI, it seems clear that only one profile can be activated at any time, not multiple profiles (either you have the default profile activated or some other being either something you manually activated or that the loader decided that should be activated).

So far, up to now, every time I activated a profile their configurations came in effect immediately, and configuration the previous profile specified were discarded.

For example:

1. Configure your default profile to use eSpeak American English voice.
2. Create another profile, say test1, and configure it to use British English Voice.
3. Create another profile, say test2, and do not change the voice settings.

Now, do the following:

1. Activate the default probile and read a word. You should hear the American English Voice.
2. Now, activate the test1 profile manually and read the same word. You should hear it using the British English Voice.
3. Now, Activate the test2 profile, that one you didn't supply voice configuration for. Read the word again, and you will hear it in American English voice.


If, as you stated, there could be multiple profiles active, you would hear the word read in British English Voice, because test2 didn't have voice specific configurations and the profile activated before that, called Test1, has British English as its voice.

Instead, what happened was that, when you activated test2, the test1 profile was immediately unloaded and, as test2 has no voice specific settings, the configuration on the default profile was then used.

This seems to confirm my impression that there may be only one active profile on a given moment, with the default profile being used as base for resolving settings that the active profile didn't specify.

The fact that a stack is chosen to achieve this implementation is an internal topic, and this is why I created a higher level abstraction that will give the caller the *** currently ** active profile, if there is one.

Sorry for the long text, and of course you are the maintainer here so I will end up naming the function whatever you decide, but I can not think on a better name, so if you could please provide it I will happily apply the change.

Reply:
As you demonstrated in your example, you cannot stack two profiles activated manually.
But you can stack 3 profiles (including default) in the following situation:
* Profile 0: Default profile with English TTS and no  indent reporting.
* Profile 1: Application profile, e.g. Notepad++ profile with indent reporting; this profile is triggered when Notepad++ is focused
* Profile 2: Manual profile with voice configured to Spanish TTS. And with a keyboard shortcut to enable it manually.
Now if you go to Notepad++ and press the shortcut to activate the manual profile (profile 2), you will have both indent reported (Notepad++ profile option) and Spanish voice (manual profile option).

Reply:
You have a good point here.

This is getting confusing, because now NVDA behavior is not deterministic based on a given profile, since the stack below it can not be predicted. Anyways, I can not extend this to the dictionaries, because then one would never know exactly what to expect on pronunciation and substitution rules. I will delete this convenience function and access the top of stack directly when needed.